cep47-logic/src/tests.rs

```diff
@@ -182,59 +176,36 @@ impl CEP47Storage for TestStorage {
             self.total_supply = self.total_supply - 1;
         }
         self.balances.insert(owner.clone(), owner_new_balance);
-        self.tokens.insert(owner, owner_new_tokens);
+        self.tokens.insert(*owner, owner_new_tokens);
     }
 
-    fn burn_one(&mut self, owner: PublicKey, token_id: TokenId) {
-        let owner_tokens = self.tokens.get(&owner);
-        let owner_balance = self.balances.get(&owner);
-        let owner_new_balance = if owner_balance.is_none() {
-            U256::from(0)
-        } else {
-            owner_balance.unwrap().clone()
-        };
-        let mut owner_new_tokens = if owner_tokens.is_none() {
-            Vec::<TokenId>::new()
-        } else {
-            owner_tokens.unwrap().clone()
-        };
-        let index = owner_new_tokens
-            .iter()
-            .position(|x| *x == token_id.clone())
-            .unwrap();
-        owner_new_tokens.remove(index);
-        self.token_metas.remove(&token_id.clone());
-        self.belongs_to.remove(&token_id.clone());
-        self.total_supply = self.total_supply - 1;
-        self.balances.insert(owner.clone(), owner_new_balance - 1);
-        self.tokens.insert(owner, owner_new_tokens);
+    fn update_token_metadata(&mut self, token_id: &TokenId, meta: Meta) {
+        self.token_metas.insert(token_id.clone(), meta).unwrap();
     }
 
-    fn new_uref(&mut self, token_id: TokenId) -> Option<URef> {
-        let mut rng = rand::thread_rng();
-        let val: [u8; 32] = rng.gen();
-        let uref = URef::new(val, AccessRights::READ_ADD_WRITE);
-        if self.urefs.contains_key(&uref) {
-            None
-        } else {
-            self.urefs.insert(uref, token_id);
-            Some(uref)
+    fn gen_token_ids(&mut self, n: u32) -> Vec<TokenId> {
+        let mut tokens = Vec::new();
+        for _ in 0..n {
+            let id = format!("token_{}", &self.token_id_generator);
+            tokens.push(id);
+            self.token_id_generator += 1;
         }
+        tokens
     }
 
-    fn del_uref(&mut self, token_uref: URef) -> Option<TokenId> {
-        let token_id = self.token_id(token_uref);
-        if token_id.is_none() {
-            None
-        } else {
-            let token_id = token_id.unwrap();
-            self.urefs.remove(&token_uref);
-            Some(token_id)
+    fn validate_token_ids(&self, token_ids: &Vec<TokenId>) -> bool {
+        for token_id in token_ids {
+            if self.onwer_of(token_id).is_some() {
```

Comment:
owner? /*not a breaking problem*/

Reply:
That's the cheapest way to check if the the token exists.

Reply:
I meant that it's misspelled onwer -> owner